lib/Service/OpenAiAPIService.php

```diff
@@ -367,7 +372,7 @@ public function createChatCompletion(
 			$usage = $response['usage']['total_tokens'];
 
 			try {
-				$this->quotaUsageMapper->createQuotaUsage($userId, Application::QUOTA_TYPE_TEXT, $usage);
+				$this->quotaUsageMapper->createQuotaUsage($userId ?? '', Application::QUOTA_TYPE_TEXT, $usage);
 			} catch (DBException $e) {
 				$this->logger->warning('Could not create quota usage for user: ' . $userId . ' and quota type: ' . Application::QUOTA_TYPE_TEXT . '. Error: ' . $e->getMessage(), ['app' => Application::APP_ID]);
```

Comment:
I think you should return from the function if the userId is null since it will not be called from a command at any point in time apparently. And maybe return after catch as well? This seems very permissive in terms of quota handling, or maybe it is intended to be?

Reply:
Currently, if a task is scheduled to run in the background it will not have an associated session and userId would be null. Thus, this ensures that the endpoint cannot be overused by scheduling a task through OCS if quota is enforced. There is a PR for core by Marcel that provides a userId even for scheduled tasks but it's still WIP.

Reply:
Ah, thanks for the explanation!